pkg/registry/core/namespace/storage/storage_test.go

```diff
@@ -189,6 +189,91 @@ func TestDeleteNamespaceWithCompleteFinalizers(t *testing.T) {
 	}
 }
 
+func TestDeleteWithGCFinalizers(t *testing.T) {
+	storage, server := newStorage(t)
+	defer server.Terminate(t)
+	defer storage.store.DestroyFunc()
+
+	propagationBackground := metav1.DeletePropagationBackground
+	propagationForeground := metav1.DeletePropagationForeground
+	propagationOrphan := metav1.DeletePropagationOrphan
+
+	var tests = []struct {
+		name          string
+		deleteOptions *metav1.DeleteOptions
+		// remaining finalizers after deletion call
+		finalizers map[string]bool
+	}{
+		{
+			name:          "propagation-nil",
+			deleteOptions: nil,
+			finalizers:    map[string]bool{},
+		},
+		{
+			name: "propagation-background",
+			deleteOptions: &metav1.DeleteOptions{
+				PropagationPolicy: &propagationBackground,
+			},
+			finalizers: map[string]bool{},
+		},
+		{
+			name: "propagation-foreground",
+			deleteOptions: &metav1.DeleteOptions{
+				PropagationPolicy: &propagationForeground,
+			},
+			finalizers: map[string]bool{
+				metav1.FinalizerDeleteDependents: true,
+			},
+		},
+		{
+			name: "propagation-orphan",
+			deleteOptions: &metav1.DeleteOptions{
+				PropagationPolicy: &propagationOrphan,
+			},
+			finalizers: map[string]bool{
+				metav1.FinalizerOrphanDependents: true,
+			},
+		},
+	}
+
+	for _, test := range tests {
+		key := "namespaces/" + test.name
+		ctx := genericapirequest.NewContext()
+		namespace := &api.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: test.name,
+				Finalizers: []string{
+					metav1.FinalizerOrphanDependents,
+					metav1.FinalizerDeleteDependents,
+				},
+			},
+			Spec: api.NamespaceSpec{
+				Finalizers: []api.FinalizerName{},
+			},
+			Status: api.NamespaceStatus{Phase: api.NamespaceActive},
+		}
+		if err := storage.store.Storage.Create(ctx, key, namespace, nil, 0, false); err != nil {
+			t.Fatalf("unexpected error: %v", err)
+		}
+		if obj, _, err := storage.Delete(ctx, test.name, test.deleteOptions); err != nil {
+			t.Errorf("unexpected error: %v", err)
+		} else {
```

Comment:
Use t.Fatal in the first case and remove the "else" here to reduce indention.

Reply:
Sounds good.